appendices/VK_EXT_physical_device_drm.adoc

```diff
@@ -25,12 +25,19 @@ Unlike the EGL extension, this extension does not expose a string containing
 the name of the device file and instead exposes device minor numbers.
 
 DRM defines multiple device node types.
-Each physical device may have one primary node and one render node
-associated.
+Each physical device is associated with a single underlying DRM device, which
+may have one primary node and one render node.
+The primary node, in the context of this extension, refers to the capability of
+that device node to be used as a display subsystem.
```

Comment:
Is this saying you can return primary node and render node from separate DRM devices on the same Vulkan physical device or not? It's not clear to me.

Reply:
In this snippet I just meant to clarify the terminology for the context of the extension. As in "VK_EXT_physical_device_drm primary node" != "DRM primary node". This is also due to the existing sentence which says `Physical devices may have no primary node (e.g. if the device does not have a display subsystem)`.

For the overall extension I want to say that it should not mix and return primary node and render node from separate DRM devices on the same Vulkan physical device.

I'm open to suggestions for phrasing this the best way possible.

Reply:
The version is tracked in the XML file, which generates content in vulkan_core.h.

Reply:
Yeah, I'm pretty comfortable with the suggested language; users can use libdrm to map between render nodes and primary nodes of the same device.

Reply:
To me, the important bit is to forbid advertising two separate DRM devices with this extension. I don't have a strong preference regarding the primary device being exposed only if the device is KMS-capable. Some notes:

- If the device is not KMS-capable, the primary node is fairly useless.
- In general, applications can't expect to have the permission to open the primary node.

> The primary node, in the context of this extension, refers to the capability of that device node to be used as a display subsystem.

That's a bit confusing IMHO. Maybe instead of trying to override what having a primary node means, we could add "If the DRM device is KMS-capable" elsewhere in the extension?

Reply:
I guess I have a slightly stronger opinion than I thought here. Right now our logic to deduce the primary node and advertise it here doesn't have to know whether there's a display subsystem on the GPU in question. If we have to go and toggle that logic based on whether there's a display subsystem associated with the GPU, which is slightly more than a trivial amount of work in the driver.

So in other words, I'm fine with not advertising two separate DRM devices in primary and render, but I'd have a preference for not requiring drivers to hide primary nodes for displayless GPUs, because it's not what our driver does now and because it's non-trivial to switch to that behavior. Looking at it separately from the principle of least surprise POV, I agree the primary node isn't necessarily that useful on displayless GPUs, and apps can't always access it, but it's there. It seems a little weird to hide it in Vulkan when it isn't hidden on the filesystem.

That said, if everyone is set on the idea of hiding it, I could be persuaded. I'd probably want some time to go make sure there aren't any larger hurdles than anticipated finding the "has display" boolean early enough to set this field up properly though before giving my final thumbs up.

Reply:
In the last (current as of now) update to this PR, I dropped the part about hiding the primary device. After the first round of comments I also realized that explicitly hiding it only makes things more complicated for little benefit.

The current proposal is to just expose the primary node of the render device anyway since it is there. So the focus of the change now is to just expliticly write to not advertise nodes from two different DRM devices.